Orange/data/table.py

```diff
@@ -112,26 +112,32 @@ def __setitem__(self, key, value):
                 self.table.metas[self.row_index, -1 - key] = value
 
     def _str(self, limit):
-        def sp_values(matrix, variables):
+        def sp_values(matrix, variables, show_labels=True):
             if not sp.issparse(matrix):
                 return Instance.str_values(matrix[row], variables, limit)
-            begptr, endptr = matrix.indptr[row:row + 2]
-            rendptr = endptr if not limit else min(endptr, begptr + 5)
-            variables = [variables[var]
-                         for var in matrix.indices[begptr:rendptr]]
+
+            # Sparse matrices can't handle slices where an index is out of
+            # bounds like numpy arrays can, so we must determine largest index
+            _, columns = matrix.shape
+            max_idx, has_more_columns = min(5, columns), columns > 5
+
             s = ", ".join(
-                "{}={}".format(var.name, var.str_val(val))
-                for var, val in zip(variables, matrix.data[begptr:rendptr]))
-            if limit and rendptr != endptr:
+                ("%s=" % var.name if (var.is_discrete or matrix[row, idx]) and
+                                     show_labels else "")
+                + var.str_val(matrix[row, idx])
+                for var, idx in zip(variables, range(max_idx)))
+
+            if limit and has_more_columns:
                 s += ", ..."
+
             return s
 
         table = self.table
         domain = table.domain
         row = self.row_index
         s = "[" + sp_values(table.X, domain.attributes)
         if domain.class_vars:
-            s += " | " + sp_values(table._Y, domain.class_vars)
+            s += " | " + sp_values(table._Y, domain.class_vars, False)
```

Comment:
IMO this isn't OK if there are multiple variables. For example, pass some text data through BoW and move all features to class variables. Then in the print, I only get values, and there is no way to know which value corresponds to which variable.

<img width="1440" alt="screen shot 2017-07-10 at 16 21 58" src="https://user-images.githubusercontent.com/713026/28022787-6a3c4be2-658c-11e7-80d4-e7996a53f519.png">


I agree that we can probably skip the label if there is only one sparse variable set as class, but we cannot skip it when there are multiple.